Source/ASTextNode.mm

```diff
@@ -1133,36 +1160,47 @@ - (UIEdgeInsets)shadowPaddingWithRenderer:(ASTextKitRenderer *)renderer
 
 - (void)setTruncationAttributedText:(NSAttributedString *)truncationAttributedText
 {
-  ASDN::MutexLocker l(__instanceLock__);
-  
-  if (ASObjectIsEqual(_truncationAttributedText, truncationAttributedText)) {
-    return;
+  {
+    ASDN::MutexLocker l(__instanceLock__);
+    
+    if (ASObjectIsEqual(_truncationAttributedText, truncationAttributedText)) {
+      return;
+    }
+
+    _truncationAttributedText = [truncationAttributedText copy];
   }
 
-  _truncationAttributedText = [truncationAttributedText copy];
   [self _invalidateTruncationText];
```

Comment:
Since you're auditing this file:
- `_invalidateTruncationText` calls `_updateComposedTruncationText` which seems to manage the lock itself. Could you please audit [this](https://github.com/facebook/AsyncDisplayKit/pull/3159/files#diff-271d6b828178013b300f794e8ec8c740R404). 
- Should `_prepareTruncationStringForDrawing` and `_composedTruncationText` be `_locked_`?